---
title: "Ketone Tolerance in CD/KD A/J Mice"
author: "Cody Cousineau, Trey Carr, and Dave Bridges"
date: "August 8, 2019"
output:
  html_document:
    toc: yes
    keep_md: yes
---

```{r setup, include=FALSE}
library(knitr)
#figures makde will go to directory called figures, will make them as both png and pdf files 
opts_chunk$set(fig.path='figures/',
               echo=TRUE, warning=FALSE, message=FALSE,dev=c('png','pdf'))
options(scipen = 2, digits = 3)
# set echo and message to TRUE if you want to display code blocks and code output respectively

knitr::knit_hooks$set(inline = function(x) {
  knitr:::format_sci(x, 'md')
})


superpose.eb <- function (x, y, ebl, ebu = ebl, length = 0.08, ...)
  arrows(x, y + ebu, x, y - ebl, angle = 90, code = 3,
  length = length, ...)

  
se <- function(x) sd(x, na.rm=T)/sqrt(length(x))

#load these packages, nearly always needed
library(tidyr)
library(dplyr)

# sets maize and blue color scheme
color.scheme <- c('#00274c', '#ffcb05')
```

This script was most recently run on `r date()` and can be found in `r getwd()`.

# Purpose

Analyse mRNA expression from AJ Muscles

# Experimental Details

qPCR for several ketolytic genes, done on three separate days

# Raw Data


```{r data-entry}
library(readr)
aj.mapping.file <- 'AJ Mapping Data.csv'
run.1.file  <- 'Muscle qPCR Run 1.csv'
run.2.file <- 'Muscle qPCR Run 2.csv'
run.3.file <- 'Muscle qPCR Run 3.csv'

mapping.data <- read_csv(aj.mapping.file)
run.1.data <- read_csv(run.1.file,na="Undetermined") %>% mutate(Plate="First") %>% filter(Target !="Rplp0")
run.2.data <- read_csv(run.2.file,na="Undetermined") %>% mutate(Plate="Second") %>% filter(!(Well %in% c("B1","A12")))
run.3.data <- read_csv(run.3.file,na="Undetermined") %>% mutate(Plate="Third")

run.data <- bind_rows(run.1.data,run.2.data,run.3.data) %>%
  left_join(mapping.data,by=c("Sample"="Mouse"))
```

The mapping file is in `r aj.mapping.file` and the two qPCR runs are in `r run.1.file` and `r run.2.file`

Why is there six replicates of some samples, like 4102

# Analysis

# Quality Check

Rearranged data so that replicates are in long format

```{r quality-check}
qpcr.data.unclean <- 
  run.data %>%
  select(Diet,Cohort,Sex,Sample,Target,Cq) %>%
  group_by(Diet,Cohort,Sex,Sample,Target) %>%
  summarize(Ct.mean = mean(Cq,na.rm=T),
            Ct.range = max(Cq,na.rm=T)-min(Cq,na.rm=T))

# tech.outliers <- arrange(qpcr.data.unclean,-Ct.range)
# tech.outliers %>% kable(caption="Technical Outliers")

# run.data %>%
#   filter(Target=="OXCT1",Sample=="4102") %>%
#   select(Cq,Well,Plate)


run.data.clean <- filter(run.data,!(Well=="H9"&Plate=="First"),
                         !(Well=="E3"&Plate=="Second"))

qpcr.data <-
  run.data.clean %>%
  select(Diet,Cohort,Sex,Sample,Target,Cq) %>%
  group_by(Diet,Cohort,Sex,Sample,Target) %>%
  summarize(Ct.mean = mean(Cq,na.rm=T),
            Ct.range = max(Cq,na.rm=T)-min(Cq,na.rm=T))

```

# Normalization

```{r norm}
analysed.data <-
  qpcr.data %>%
  ungroup %>%
  filter(!(Sample %in% c('4106','4108'))) %>%
  group_by(Diet,Cohort,Sex,Sample) %>%
  mutate(Ct.norm = Ct.mean-Ct.mean[Target=="Rplp0"]) %>%
  group_by(Diet,Sex,Target,Sample) %>%
  mutate(Ct.quant = 2^-Ct.norm) 

summarized.data <-
  analysed.data %>%
  group_by(Diet,Sex,Target) %>%
  summarise(Mean.exp = mean(Ct.quant),
            Mean.se = se(Ct.quant)) %>%
  group_by(Target) %>%
  mutate(Mean.rel = Mean.exp/Mean.exp[Diet=="Control"&Sex=="F"],
         Mean.rel.se = Mean.se/Mean.exp[Diet=="Control"&Sex=="F"]) 
```

## Quality Check

```{r quality check}
tech.outliers <- arrange(qpcr.data,-Ct.range)
tech.outliers %>% 
  filter(Ct.range>2,Ct.mean<30) %>%
  kable(caption="Technical Outliers")

#biological outliers
analysed.data %>% 
  group_by(Diet,Sex,Target) %>%
  mutate(Group.mean=mean(Ct.quant),
         Diff.from.mean = Ct.quant-mean(Ct.quant)) %>%
  arrange(-abs(Diff.from.mean)) %>% 
  head %>% 
  kable
  
```

# BDH1

```{r bdh1-muscle-expression}
library(ggplot2)
ggplot(summarized.data %>% filter(Target=="BDH1"),
       aes(y=Mean.rel,
           ymin=Mean.rel-Mean.rel.se,
           ymax=Mean.rel+Mean.rel.se,
           x=Sex,
           fill=Diet)) +
  geom_bar(stat='identity',position="dodge",width=0.75) +
  geom_errorbar(width = .5, position = position_dodge(0.75)) +
  labs(title="Bdh1",
       y="Relative Expression",
       x="")
```

# OXCT1

```{r oxct1-muscle-expression}
ggplot(summarized.data %>% filter(Target=="OXCT1"),
       aes(y=Mean.rel,
           ymin=Mean.rel-Mean.rel.se,
           ymax=Mean.rel+Mean.rel.se,
           x=Sex,
           fill=Diet)) +
  geom_bar(stat='identity',position="dodge",width=0.75) +
  geom_errorbar(width = .5, position = position_dodge(0.75)) +
  labs(title="Oxct1",
       y="Relative Expression",
       x="") -> oxct.plot

oxct.plot
```

# SLC16A1

```{r slc16a1-muscle-expression}
ggplot(summarized.data %>% filter(Target=="Slc16a1"),
       aes(y=Mean.rel,
           ymin=Mean.rel-Mean.rel.se,
           ymax=Mean.rel+Mean.rel.se,
           x=Sex,
           fill=Diet)) +
  geom_bar(stat='identity',position="dodge",width=0.75) +
  geom_errorbar(width = .5, position = position_dodge(0.75)) +
  labs(title="Slc16a1",
       y="Relative Expression",
       x="") -> slc16a1.plot

slc16a1.plot
```

# SLC16A6

```{r slc16a6-muscle-expression}
ggplot(summarized.data %>% filter(Target=="Slc16a6"),
       aes(y=Mean.rel,
           ymin=Mean.rel-Mean.rel.se,
           ymax=Mean.rel+Mean.rel.se,
           x=Sex,
           fill=Diet)) +
  geom_bar(stat='identity',position="dodge",width=0.75) +
  geom_errorbar(width = .5, position = position_dodge(0.75)) +
  labs(title="Slc16a6",
       y="Relative Expression",
       x="") -> slc16a6.plot

slc16a6.plot
```

# Summary Plots

```{r ketolytic-muscle-expression}
ggplot(summarized.data %>% filter(Target!="Rplp0"),
       aes(y=Mean.rel,
           ymin=Mean.rel-Mean.rel.se,
           ymax=Mean.rel+Mean.rel.se,
           x=Sex,
           fill=Diet)) +
  geom_bar(stat='identity',position="dodge",width=0.75) +
  geom_errorbar(width = .5, position = position_dodge(0.75)) +
  facet_grid(.~Target) +
  labs(title="Ketolytic Gene Expression",
       y="Relative Expression",
       x="") +
  scale_fill_grey() +
  theme_classic(base_size = 16) +
  theme(legend.position=c(0.9,0.8))

```

# Statistics

## Sample Size

```{r sample-size}
analysed.data %>%
  ungroup() %>%
  group_by(Sex,Diet,Target) %>%
  count %>%
  pivot_wider(names_from="Target",values_from = "n") %>%
  kable(caption="Group sizes for analyzed data")
```

## Multivariate Analysis

Did 2x2 ANOVA analyses

```{r anova}
library(broom)

interaction.results <- data.frame()
for (gene in c("BDH1","OXCT1","Slc16a1","Slc16a6")) {
temp.result <- lm(Ct.quant ~ Diet*Sex,data=filter(analysed.data, Target==gene)) %>% 
  tidy %>% 
  mutate(Gene=gene) %>% 
  filter(term=="DietKetogenic:SexM")
interaction.results <- bind_rows(interaction.results,temp.result)
}

kable(interaction.results,caption="Test for Diet-Sex Interactions")

main.results <- data.frame()
for (gene in c("BDH1","OXCT1","Slc16a1","Slc16a6")) {
temp.result <- lm(Ct.quant ~ Diet+Sex,data=filter(analysed.data, Target==gene)) %>% 
  tidy %>% 
  mutate(Gene=gene) 
main.results <- bind_rows(main.results,temp.result)
}

main.results.formatted <- 
  main.results %>% 
  relocate(Gene,.before=term) %>%
  select(-statistic) %>%
  group_by(Gene) %>%
  mutate(Pct.Change = estimate/estimate[term=="(Intercept)"]*100) %>%
  mutate(Sig=case_when(p.value<0.05&term!="(Intercept)"~"*",
                       .default=""))

kable(main.results.formatted,caption="Test for Diet and Sex Effects on Expression")
```

```{r pairwise-analyses}
library(car)
analysed.data %>%
  group_by(Target,Sex) %>%
  filter(Target!="Rplp0") %>%
  summarize(Pct.Decrease = (mean(Ct.quant[Diet=="Control"],na.rm=T)-mean(Ct.quant[Diet=="Ketogenic"],na.rm=T))/mean(Ct.quant[Diet=="Control"],na.rm=T)*100,
            Shapiro=min(shapiro.test(Ct.quant[Diet=="Ketogenic"])$p.value,
                        shapiro.test(Ct.quant[Diet=="Control"])$p.value),
            Levene=leveneTest(Ct.quant~Diet)$"Pr(>F)"[1],
            Mann.Whitney=wilcox.test(Ct.quant~Diet)$p.value,
            Welch=t.test(Ct.quant~Diet,var.equal = F)$p.value,
            Student=t.test(Ct.quant~Diet,var.equal = T)$p.value) %>%
  mutate(Test=case_when(Shapiro<0.05~"Mann Whitney",
                        Levene<0.05~"Welch's",
                        .default="Student's")) %>%
  kable(caption="Pairwise tests for effects of diet")
  

```

```{r outlier-samples}
analysed.data %>%
  filter(Target=="Slc16a1") %>%
  filter(Sex=="F") %>%
  kable(caption="Visually inspected outliers")

#Slc16a1 :: 4108 seems variable, not due to technical variation
```

# Comparing SLC16 Expression

```{r slc16a1-slc16a6-comparason}
analysed.data %>%
  ungroup %>%
  group_by(Target) %>%
  summarize(Avg=mean(Ct.quant,na.rm=T),
            SE = se(Ct.quant)) %>%
  filter(Target %in% c("Slc16a1","Slc16a6")) %>%
  mutate(Rel.Avg = Avg/min(Avg),
         Rel.SE = SE/min(Avg)) -> Slc16.summary
  
kable(Slc16.summary, caption="Relative expression of MCT isoforms")

ggplot(Slc16.summary,
       aes(y=Rel.Avg,
           ymin=Rel.Avg-Rel.SE,
           ymax=Rel.Avg+Rel.SE,
           x=Target)) +
  geom_bar(stat="identity") +
  geom_errorbar(width=0.5) +
  labs(y="Relative Expression",
       x="",
       title="MCT1 Isoforms in Quadriceps") +
  theme_classic(base_size = 16)

```

# Session Information

```{r session-info}
sessionInfo()
```

