---
title: "Combined Barplots for aTSC Mammary Gland Studies"
author: "Dave Bridges and Noura El Habbal"
date: "August 29, 2020"
output:
  html_document:
    highlight: tango
    keep_md: yes
    number_sections: yes
    toc: yes
  pdf_document:
    highlight: tango
    keep_tex: yes
    number_sections: yes
    toc: yes
---

```{r global_options, include=FALSE}
library(knitr)
#figures makde will go to directory called figures, will make them as both png and pdf files 
opts_chunk$set(fig.path='figures-noura/',
               echo=TRUE, warning=FALSE, message=FALSE,dev=c('png','pdf'))
options(scipen = 2, digits = 3)
# set echo and message to TRUE if you want to display code blocks and code output respectively

knitr::knit_hooks$set(inline = function(x) {
  knitr:::format_sci(x, 'md')
})


superpose.eb <- function (x, y, ebl, ebu = ebl, length = 0.08, ...)
  arrows(x, y + ebu, x, y - ebl, angle = 90, code = 3,
  length = length, ...)

  
se <- function(x) sd(x, na.rm=T)/sqrt(length(x))

#load these packages, nearly always needed
library(tidyr)
library(dplyr)

# sets maize and blue color scheme
color.scheme <- c('#00274c', '#ffcb05')
```

# Purpose

# Raw Data

Imported normalized counts from Salmon aligned data, analyzed via DESeq2

```{r data-input}
library(readr) #loads the readr package
counts.file <- 'DESeq2 Normalized Counts.csv'
design.file <- 'Design Table.csv'

counts.table <- read_csv(counts.file) 

library(biomaRt)
ensembl = useMart("ensembl",dataset="mmusculus_gene_ensembl")
gene.mapping <- getBM(attributes=c('mgi_symbol','ensembl_gene_id'),
                      values=counts.table$X1,
                      filters='ensembl_gene_id',
                      mart=ensembl)

annotated.counts.table <- full_join(counts.table,gene.mapping, by=c('X1'='ensembl_gene_id'))

long.counts.table <- pivot_longer(annotated.counts.table,
                                  cols=starts_with('14'),
                                  names_to="Sample",
                                  values_to="TPM")

design.table <- read_csv(design.file)
long.merged.data <- 
  design.table %>%
  dplyr::select(names, Genotype,) %>%
  mutate(Genotype= relevel(as.factor(Genotype),ref="WT")) %>%
  full_join(long.counts.table, by=c('names'='Sample'))
```

These data can be found in **`r getwd()`**.  The counts file is `r counts.file` but the design file is `r design.file`.

# Analysis

# Single Gene Graphs

```{r genes.of.interest}
#use MGI symbols and add them to this list

human.mouse.mapping.table <- 'http://www.informatics.jax.org/downloads/reports/HOM_MouseHumanSequence.rpt'
mapping.table <- read_tsv(human.mouse.mapping.table) %>%
  dplyr::select(`Common Organism Name`,Symbol,`HomoloGene ID`) 
  
wide.mapping.table <- pivot_wider(mapping.table,
                                  names_from=`Common Organism Name`,
                                  values_from = Symbol,
                                  id_cols=`HomoloGene ID`) %>%
  rename("Mouse"='mouse, laboratory') %>%
  mutate(Mouse=as.factor(as.character(Mouse))) %>%
    mutate(human=as.factor(as.character(human)))


genes.of.interest <- c('Cxcl13','Cxcr5','Ccr6','Ccr7', 'Il12a', 'Fut7', 'Tnfsf14', 'Cr2', 'Azgp1','Aicda', 'Lef1', 'Cd19', 'Cd22', 'Siglecg','Foxp3', 'Cxcr5', 'Ntrk1', 'Sstr4', 'Pla2g2d', 'Pla2g5', 'Pla2g3','Pla2g4f','Pla2g6', 'Cyp4f18', 'Agtr2', 'Fabp3', 'Zan', 'Cntnap2', 'Foxp3', 'Pax5','Il10', 'Pcdha10','Cnr2', 'Kcna1', 'Azgp1', 'Lef1', 'Cd79a','Cd3d', 'Cd247', 'Cd4', 'Cd3g', 'Cd28', 'Cd3e', 'Ccr7', 'Lck', 'Blk', 'Itk', 'Syk', 'Myoz2', 'Mypn', 'Mybph', 'Myom2', 'Myh8', 'Myh7', 'Myh4', 'Myh1', 'Myl2','Myl1', 'MYPF', 'Acta1','Ryr1','Csrp3','Prkag3', 'Tnnt2','Tnnt3', 'Acaca', 'Srebf1', 'Acly', 'Fasn', 'Rps6kb1', 'Eif4ebp1', 'Acat1','Akt1', 'Rps14', 'Plin1', 'Plin2',  'Fabp4', 'Dgat1', 'Dgat2', 'Lpl', 'Pnpla2', 'Gpat3', 'Acsl1', 'Acsl3', 'Acsl4', 'Acsl5', 'Acsl6', 'Elovl1', 'Elovl2', 'Elovl3', 'Elovl4', 'Elovl5', 'Elovl6', 'Elovl7', 'Scd1', 'Scd2', 'Scd3', 'Scd4', 'Fads1', 'Fads2','Fads2b','Fads3','Fads6', 'Mfsd2a', 'Mfsd6', 'Mfsd8', 'Mfs10', 'Mfsd14a', 'Mfsd14b', 'Cyp2d34', 'Cyp4f18', 'Cyp1a2', 'Cyp2c50', 'Cyp4a12a', 'Cyp4a12b', 'Cyp2c50') #you can choose any gene name  but by default it show the sig genes , previously this code had c('Fasn','Wap','Csn2','Insr') UNCERTAIN about CNR3 to find homologous if it is pcdha10 or not, used Pcdha10 for mouse which has different homologene ID number that that of humans; MYPF does not exist as a gene (possible typo? when indeed it is MYLF?); 
#from the DESEQ model
deseq.results.file <-  'DESeq2 Results.csv'
deseq.results <- read_csv(deseq.results.file)
sig.genes <- filter(deseq.results, padj<0.05) %>% pull(symbol)

genes.to.plot <- c(genes.of.interest)#,sig.genes)
```

```{r barplots}
library(ggplot2)
for (gene in genes.to.plot){
  barplot <- long.merged.data %>%
    filter(mgi_symbol==gene) %>%
    group_by(Genotype) %>%
    summarize(Mean = mean(TPM),
              Error =se(TPM)) %>%
    ggplot(aes(y=Mean,x=Genotype,fill=Genotype)) +
    geom_bar(stat='identity') +
    geom_errorbar(aes(ymin=Mean-Error,
                      ymax=Mean+Error),
                  width=0.5) +
    theme_bw() +
    theme(text=element_text(size=18)) +
    labs(y="Normalized Counts (TPM)",
         x="",
         title=gene)
  
  ggsave(barplot,
         device='pdf',
         filename=c(paste('figures-noura/barplots/',paste(gene,'-barplot.pdf', sep=''), sep=''))) 
  ggsave(barplot, 
         device='png',
         filename=c(paste('figures-noura/barplots/',paste(gene,'-barplot.png', sep=''), sep='')))
}
```

```{r boxplots}
for (gene in genes.to.plot){
  barplot <- long.merged.data %>%
    dplyr::filter(mgi_symbol==gene) %>%
    ggplot(aes(y=TPM,x=Genotype,col=Genotype)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(position=position_jitterdodge(jitter.width=0.25)) +
    theme_bw() +
    theme(text=element_text(size=18)) +
    labs(y="Normalized Counts (TPM)",
         x="",
         title=gene) +
    expand_limits(y=0)
  
  ggsave(barplot,
         device='pdf',
         filename=c(paste('figures-noura/boxplots/',paste(gene,'-boxplot.pdf', sep=''), sep=''))) 
  ggsave(barplot, 
         device='png',
         filename=c(paste('figures-noura/boxplots/',paste(gene,'-boxplot.png', sep=''), sep='')))
}
```

# Grouped Gene Sets

```{r gene-sets}
library(stringr)
gene.sets <- tibble(name="Fatty Acid Synthesis", genes = list('Fasn','Acly','Acaca'))
gene.sets <- 
  gene.sets %>% 
    add_row(name="Milk Proteins", genes=list('Wap','Csn1s1','Csn1s2a','Csn1s2b','Csn2','Csn3','Lalba','Ltf')) %>%
  add_row(name="T and B Cell Markers", genes=list('Cd3e','Cd3d','Cd3g','Ptprc','Cd19')) %>% #CD3, CD45(Ptprc), CD19
    add_row(name="B Cell Markers", genes=list('Leuf1','Ptprc','Cd19','Sdc1','Tnfrsf13b','Spn','Cd79a')) %>% #see https://www.novusbio.com/antibody-news/how-to-identify-b-cell-subsets-using-flow-cytometry CD45(Leufr), CD45R/B220(Ptprc), CD19, CD138(Sdc1), TACI(Tnfrsf13b), CD43(Spn)
  add_row(name="Activated B Cells", genes=list('Cd19','Il2ra','Tnfrsf8')) %>%
  add_row(name="Plasma Cells", genes=list('Cd27','Cd38','Cd78','Sdc1','Slamf7','Il6')) %>%  
    add_row(name="Memory B Cells", genes=list('Cd19','Cd27','Ms4a1','Cd40','Cd80','	Pdcd1lg2','Cxcr3','Cxcr5','Cxcr6')) %>%  
   add_row(name="T Cell Markers", genes=list('Cd8a','Cd4','Il2ra','Foxp3','Il7r')) %>% #CD8 (Killer), CD4 (helper), CD25(Il2ra)/FoxP3,CD127(Il7r) Tregs)
  add_row(name="Macrophage Markers", genes=list('Fcgr1','Ly6g','Itgam','Itgax','Csf1r')) %>% #CD64=Fcgr1,CD11B=Itgam,CD11C=Itgax, CD115=Csf1r 
  add_row(name="Phospholipases",
          genes=list(gene.mapping[grepl('Pla2g',gene.mapping$mgi_symbol),]$mgi_symbol)) %>%
 add_row(name="New Targets",
         genes=list('Thrsp','Agpat6','Plin2','Aox1','Aox2','Aox3','Aox4','Btn1a1','Btn1a2')) %>%
  add_row(name="Xanthine Oxidoreductases",
          genes=list('Aox1','Aox2','Aox3','Aox4','Xdh')) %>%
  add_row(name="DHA Receptors",
          genes=list('Gpr18','Gpr32','Cmklr1','Gpr37'))%>%
  add_row(name="LTB4 Receptors",
          genes=list('Ltb4r1','Ltb4r2')) %>%
  add_row(name="B-Cell Regulators",
          genes=list('Ccl28','Ccr10','Vcam1','Slc30a2','Ackr2')) %>% #CCL28 binds to CCR10 on B-cells (see http://dx.doi.org/10.1007/s10911-010-9188-7 and its ref45). VCAM-1 blocking decreases IgA secretion (ref 52)
    add_row(name="DMRT Targets",
          genes=list(str_to_title(c('BMI1','E2F1','EED','EZH2','JARID2','KLF4','MTF2','MYBL2','MYC','PAX3','PHC1','PPARD','RNF2','SMAD4','SMARCA4','STAT3','SUZ12','TP53','ZNF217')))) %>% # from https://maayanlab.cloud/Harmonizome/gene/DMRT2 CHEA Transcription Factor Targets
    add_row(name="PPARg Targets",
          genes=list('Plin4','Adipq','Fabp4','Cav2')) %>%
      add_row(name="Eicosanoid Signaling",
          genes=list('Plcb1', 'Ephx2','Pla2g4a', 'Ptger3','Ptgs1'))
  



for (set in gene.sets$name) {
  genes <- unlist(filter(gene.sets, name==set)$genes)
    barplot <- long.merged.data %>%
    dplyr::filter(mgi_symbol %in% genes) %>%
        group_by(mgi_symbol,Genotype) %>%
    summarize(Mean = mean(TPM),
              Error =se(TPM)) %>%
    ggplot(aes(y=Mean,
               x=factor(mgi_symbol, levels=genes),
               fill=Genotype)) +
    geom_bar(stat='identity',position='dodge',width=0.75) +
    geom_errorbar(aes(ymin=Mean-Error,
                      ymax=Mean+Error),
                  width=0.5,
                  position=position_dodge(width=0.75)) +
    scale_fill_grey() +
    scale_color_grey() +
    theme_classic() +
    theme(text=element_text(size=18),
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
          legend.position = c(0.2,0.8)) +
    labs(y="Normalized Counts (TPM)",
         x="",
         title=set) +
      scale_fill_grey()
    
  ggsave(barplot, 
         device='png',
         filename=c(paste('figures-noura/grouped-barplots/',paste(set,' Plot.png', sep=''), sep='')))
  
    ggsave(barplot, 
         device='pdf',
         filename=c(paste('figures-noura/grouped-barplots/',paste(set,' Plot.pdf', sep=''), sep='')))
}
```

# Session Information

```{r session-information, echo=T}
sessionInfo()
```