---
title: "Analysis of Secreted Proteins from TSC Knockout Muscles"
author: "Dave Bridges"
date: "March 26, 2020"
output:
  html_document:
    highlight: tango
    keep_md: yes
    number_sections: no
    toc: yes
  pdf_document:
    highlight: tango
    keep_tex: yes
    number_sections: yes
    toc: yes
---

```{r global_options, include=FALSE}
library(knitr)
#figures makde will go to directory called figures, will make them as both png and pdf files 
opts_chunk$set(fig.path='figure/',
               echo=TRUE, warning=FALSE, message=FALSE,dev=c('png','pdf'))
options(scipen = 2, digits = 3)
# set echo and message to TRUE if you want to display code blocks and code output respectively

knitr::knit_hooks$set(inline = function(x) {
  knitr:::format_sci(x, 'md')
})


superpose.eb <- function (x, y, ebl, ebu = ebl, length = 0.08, ...)
  arrows(x, y + ebu, x, y - ebl, angle = 90, code = 3,
  length = length, ...)

  
se <- function(x) sd(x, na.rm=T)/sqrt(length(x))

#load these packages, nearly always needed
library(tidyr)
library(dplyr)

# sets maize and blue color scheme
color.scheme <- c('#00274c', '#ffcb05')

library("knitcitations")
cleanbib()
options("citation_format" = "pandoc")
```

# Purpose

To broadly evaluate all potential myokines from mTORC1 activated muscles based on transcriptional changes from our RNAseq data 

# Experimental Details

Used RNAseq data compiled from our previous experiments  Used biomart to extract proteins with annotated signal peptides done via SignalP `r citep('10.1038/s41587-019-0036-z')`.

# Raw Data

```{r data-input}
library(readr) #loads the readr package
rnaseq.filename <- '../data/processed/Binary DESeq Results.csv' #make this a separate line, you can use any variable you want
rnaseq.analysed.filename <- '../data/processed/Binary Normalized Counts.csv'

#this loads whatever the file is into a dataframe called exp.data if it exists
rnaseq.stats <- read_csv(rnaseq.filename)
rnaseq.counts <- read_csv(rnaseq.analysed.filename)
```

## Annotated Proteins with Signal Peptides

```{r signalp-data}
library(biomaRt)

mouse.data <- useDataset('mmusculus_gene_ensembl', mart=useMart('ensembl'))
#listFilters(mouse.data) #to locate with_signalp
#listAttributes(mouse.data)  #to locate ensembl_gene_id

signalp.data <- 
  getBM(attributes=c('ensembl_gene_id','signalp'), 
      values = rnaseq.stats$Row.names, 
      mart = mouse.data) %>%
  mutate(signap = as.factor(signalp))

signalp.genes <-
  signalp.data %>%
  filter(signalp %in% c('SignalP-noTM')) %>%
  pull(ensembl_gene_id)

membrane.genes <- getBM(attributes=c('ensembl_gene_id', 'go_id'),
                        filters = 'go', 
                        values = 'GO:0016020',
                        mart = mouse.data) %>%
  pull(ensembl_gene_id)


secreted.genes <-
  rnaseq.stats %>%
  mutate(SignalP=Row.names %in% signalp.genes,
         GO_Membrane=Row.names %in% membrane.genes) %>%
  mutate(Secreted=if_else(SignalP==TRUE&GO_Membrane==FALSE, 'Secreted','Non-Secreted'))
  
secreted.genes %>%  
  count(Secreted) %>%
  kable(caption="Predicted secreted gene products")
```

These data can be found in **`r getwd()`**.  The normalized RNAseq data can be found in a file named **`r rnaseq.filename`**.  This script was most recently updated on **`r date()`**.


```{r human-secretome}
human.secretome.link <- 'https://www.proteinatlas.org/search/sa_location%3ASecreted+to+blood?format=tsv'
human.secretome <- read_delim(human.secretome.link, delim='\t')
human.secretome.genes <-
  human.secretome %>%
  dplyr::select(Gene,`Protein class`, `Ensembl`)

human.data = useMart("ensembl", dataset = "hsapiens_gene_ensembl")

protein.atlas.genes <- getLDS(attributes = c("hgnc_symbol"), 
    filters = "hgnc_symbol", values = human.secretome$Gene, mart = human.data, 
    attributesL = c("external_gene_name"), martL = mouse.data) %>%
  distinct(Gene.name) %>%
  pull(Gene.name)
```

The human secretome was described at https://www.proteinatlas.org/search/sa_location%3ASecreted+to+blood?format=tsv and in `r citet('10.1126/science.1260419')`.  These data were downloaded from `r human.secretome.link`.  After matching to the mouse genome via biomart and removing duplicates there were `r length(protein.atlas.genes)` secreted mouse genes.

## Comparason of Secretomes

```{r secretome-venn}
library(VennDiagram)
signal.p.genes <- secreted.genes %>% 
  filter(Secreted=='Secreted') %>% 
  distinct(external_gene_name) %>% 
  pull(external_gene_name)

grid.newpage()
draw.pairwise.venn(length(protein.atlas.genes),
                                length(signal.p.genes),
                                length(intersect(protein.atlas.genes, signal.p.genes)),
                   c("`Protein Atlas`", "`SignalP Non-Membrane`"),
                   fill=color.scheme)
```

# Analysis

The ENSEMBL dataset with genes annotated as having a signalP annotation includes `r length(signalp.genes)` and `r length(membrane.genes)` that are filtered with the GO annotation of 0016020, cellular component - membrane.  This resulted in `r secreted.genes %>% filter(Secreted=="Secreted") %>% count` secreted genes.


```{r potential-secreted-proteins}
rnaseq.secreted <-
  rnaseq.stats %>%
  filter(Row.names %in% signalp.genes)  %>% #keep proteins with signal peptide
  filter(!(Row.names %in% membrane.genes))  #remove membrane proteins

rnaseq.secreted <-
  rnaseq.stats %>%
  filter(external_gene_name %in% intersect(protein.atlas.genes, signal.p.genes))    #overlap of both

rnaseq.secreted <-
  rnaseq.stats %>%
  filter(external_gene_name %in% protein.atlas.genes)   #only protein atlas


secreted.output.file <- '../data/processed/Potential Secreted Genes.csv'

rnaseq.secreted %>%
  rename('ENSEMBL Gene ID'='Row.names',
         'Gene Name'='external_gene_name') %>%
  dplyr::select(-X1) %>%
  dplyr::select('Gene Name', everything()) %>%
  arrange(-abs(log2FoldChange)) %>%
  write_csv(secreted.output.file)

library(ggplot2)

sig.secreted <- 
  rnaseq.secreted %>%
  mutate(FC = 2^(log2FoldChange)) %>%
  filter(padj < 0.05,
         baseMean>2) %>%
  arrange(desc(abs(log2FoldChange))) 

sig.secreted %>%
  head(10) %>%
  kable(caption = "Top differentially expressed secreted proteins")

gdf15.data <- filter(rnaseq.secreted, external_gene_name=="Gdf15")  
ggplot(rnaseq.secreted,
       aes(x=log2FoldChange,
           y=-log10(padj))) +
  geom_point(aes(col=padj>0.05)) +
  labs(y="P-value (-log10)",
       x="mRNA Fold Change (log2)",
       title="Differential Expression of Secreted Genes") +
  geom_segment(
     xend = gdf15.data$log2FoldChange-0.05, yend = -log10(gdf15.data$padj)+1,
     x=5, y=25,
     arrow=arrow(length=unit(0.1,'cm'))) +
  geom_text(label="Gdf15", x=5, y=27) +
  scale_color_grey() +
  theme_classic() +
  theme(text=element_text(size=18),
        legend.position='none')
```
These differentially expressed secreted proteins can be found in `r secreted.output.file`. 

Among `r rnaseq.stats %>% filter(padj<0.05) %>% count()` genes that were differentially expressed in *Tsc1* knockout muscles, `r rnaseq.secreted %>% filter(padj<0.05) %>% count()` were potential secreted proteins.

Gdf15 was upregulated `r sig.secreted %>% filter(external_gene_name=='Gdf15') %>% pull(FC)` fold (p=`r sig.secreted %>% filter(external_gene_name=='Gdf15') %>% pull(padj)`)

# Session Information

```{r session-information, echo=T}
sessionInfo()
```

# Bibliography

```{r bibliography, results='asis'}
write.bibtex(file="secreted-database-references.bib")
bibliography("markdown")
```